Sources/OpenAPIRuntime/Interface/Body.swift

```diff
@@ -0,0 +1,535 @@
+    ///     to accumulate in memory before it throws an error.
+    /// - Throws: `TooManyBytesError` if the the body contains more
+    ///   than `maxBytes`.
+    public func collectAsString(upTo maxBytes: Int) async throws -> String {
+        let bytes: DataType = try await collect(upTo: maxBytes)
+        return String(decoding: bytes, as: UTF8.self)
+    }
+}
+
+// MARK: - Body conversions
+
+extension Body: ExpressibleByStringLiteral {
+
+    public convenience init(stringLiteral value: String) {
+        self.init(data: value)
+    }
+}
+
+extension Body {
+
+    public convenience init(data: [UInt8]) {
+        self.init(data: data[...])
+    }
+}
+
+extension Body: ExpressibleByArrayLiteral {
+
+    public typealias ArrayLiteralElement = UInt8
+
+    public convenience init(arrayLiteral elements: UInt8...) {
+        self.init(data: elements)
+    }
+}
+
+extension Body {
+
+    public convenience init(data: Data) {
+        self.init(data: ArraySlice(data))
+    }
+
+    /// Accumulates the full body in-memory into a single buffer
+    /// up to `maxBytes`, converts it to Foundation.Data, and returns it.
+    /// - Parameters:
+    ///   - maxBytes: The maximum number of bytes this method is allowed
+    ///     to accumulate in memory before it throws an error.
+    /// - Throws: `TooManyBytesError` if the the body contains more
+    ///   than `maxBytes`.
+    public func collectAsData(upTo maxBytes: Int) async throws -> Data {
+        let bytes: DataType = try await collect(upTo: maxBytes)
+        return Data(bytes)
+    }
+}
+
+// MARK: - Underlying async sequences
+
+extension Body {
+
+    /// Async iterator of both input async sequences and of the body itself.
+    public struct Iterator: AsyncIteratorProtocol {
+
+        public typealias Element = Body.DataType
+
+        private let produceNext: () async throws -> Element?
+
+        init<Iterator: AsyncIteratorProtocol>(
+            _ iterator: Iterator
+        ) where Iterator.Element == Element {
+            var iterator = iterator
+            self.produceNext = {
+                try await iterator.next()
+            }
+        }
+
+        public func next() async throws -> Element? {
+            try await produceNext()
+        }
+    }
+}
+
+extension Body {
+
+    /// A type-erased async sequence that wraps input sequences.
+    private struct BodySequence: AsyncSequence {
+
+        typealias AsyncIterator = Body.Iterator
+        typealias Element = DataType
+
+        private let produceIterator: () -> AsyncIterator
+
+        init<S: AsyncSequence>(_ sequence: S) where S.Element == Element {
+            self.produceIterator = {
+                .init(sequence.makeAsyncIterator())
+            }
+        }
+
+        func makeAsyncIterator() -> AsyncIterator {
+            produceIterator()
+        }
+    }
+
+    /// A wrapper for a sync sequence.
+    private struct WrappedSyncSequence<S: Sequence>: AsyncSequence
```

Comment:
FWIW we have this in async-algorithms.

Reply:
https://github.com/apple/swift-openapi-runtime/pull/46#discussion_r1314597162